isis/src/clipper/objs/ClipperPushBroomCamera/ClipperPushBroomCamera.cpp

```diff
@@ -0,0 +1,136 @@
+/** This is free and unencumbered software released into the public domain.
+
+The authors of ISIS do not claim copyright on the contents of this file.
+For more details about the LICENSE terms and the AUTHORS, you will
+find files of those names at the top level of this repository. **/
+
+/* SPDX-License-Identifier: CC0-1.0 */
+
+#include "ClipperPushBroomCamera.h"
+
+#include "CameraDistortionMap.h"
+#include "CameraFocalPlaneMap.h"
+#include "iTime.h"
+#include "LineScanCameraGroundMap.h"
+#include "LineScanCameraSkyMap.h"
+#include "NaifStatus.h"
+#include "LineScanCameraDetectorMap.h"
+
+namespace Isis {
+  /**
+   * Constructs a ClipperPushBroomCamera object using the image labels.
+   *
+   * @param Cube &cube Clipper EIS image.
+   */
+   ClipperPushBroomCamera::ClipperPushBroomCamera(Cube &cube) : LineScanCamera(cube) {
+
+     m_spacecraftNameLong = "Europa Clipper";
+     m_spacecraftNameShort = "Clipper";
+
+     int frameCode = naifIkCode();
+
+     // ik values for NAC: -159101(general), -159110(rad), -159112(koz), filters between -159121 and -159131
+     if (frameCode == -159101 || frameCode == -159110 || frameCode == -159112
+                              || (frameCode > -159132 && frameCode < -159120)) {
+       m_instrumentNameLong  = "Europa Imaging System Push Broom Narrow Angle Camera";
+       m_instrumentNameShort = "EIS-PBNAC";
+     }
+     // ik values for WAC: -159102(general), -159111(rad), -159113(koz), filters between -159141 and -159151
+     else if (frameCode == -159102 || frameCode == -159111 || frameCode == -159113
+                                   || (frameCode > -159152 && frameCode < -159140)) {
+       m_instrumentNameLong  = "Europa Imaging System Push Broom Wide Angle Camera";
+       m_instrumentNameShort = "EIS-PBWAC";
+     }
```

Comment:
With the current kernel set we should only be looking for `frameCode == -159103` for the NAC and `frameCode == -159104` for WAC

Reply:
These have been updated.